src/modules/fancyzones/lib/JsonHelpers.cpp

```diff
@@ -0,0 +1,951 @@
+        result.SetNamedValue(L"type", json::value(TypeToString(zoneSet.type)));
+        if (zoneSet.type != ZoneSetLayoutType::Custom)
+        {
+            result.SetNamedValue(L"zone-count", json::value(*zoneSet.zoneCount));
+        }
+
+        return result;
+    }
+
+    std::optional<ZoneSetData> ZoneSetData::FromJson(const json::JsonObject& zoneSet)
+    {
+        try
+        {
+            ZoneSetData zoneSetData;
+
+            zoneSetData.uuid = zoneSet.GetNamedString(L"uuid");
+            zoneSetData.type = TypeFromString(std::wstring{ zoneSet.GetNamedString(L"type") });
+            if (zoneSetData.type != ZoneSetLayoutType::Custom)
+            {
+                zoneSetData.zoneCount = static_cast<int>(zoneSet.GetNamedNumber(L"zone-count"));
+            }
+
+            return zoneSetData;
+        }
+        catch (const winrt::hresult_error&)
+        {
+            return std::nullopt;
+        }
+    }
+
+    json::JsonObject AppZoneHistoryJSON::ToJson(const AppZoneHistoryJSON& appZoneHistory)
+    {
+        json::JsonObject result{};
+
+        result.SetNamedValue(L"app-path", json::value(appZoneHistory.appPath));
+        result.SetNamedValue(L"zoneset-uuid", json::value(appZoneHistory.data.zoneSetUuid));
+        result.SetNamedValue(L"zone-index", json::value(appZoneHistory.data.zoneIndex));
+
+        return result;
+    }
+
+    std::optional<AppZoneHistoryJSON> AppZoneHistoryJSON::FromJson(const json::JsonObject& zoneSet)
+    {
+        try
+        {
+            AppZoneHistoryJSON result;
+
+            result.appPath = zoneSet.GetNamedString(L"app-path");
+            result.data.zoneSetUuid = zoneSet.GetNamedString(L"zoneset-uuid");
+            result.data.zoneIndex = static_cast<int>(zoneSet.GetNamedNumber(L"zone-index"));
+
+            return result;
+        }
+        catch (const winrt::hresult_error&)
+        {
+            return std::nullopt;
+        }
+    }
+
+    json::JsonObject DeviceInfoJSON::ToJson(const DeviceInfoJSON& device)
+    {
+        json::JsonObject result{};
+
+        result.SetNamedValue(L"device-id", json::value(device.deviceId));
+        result.SetNamedValue(L"active-zoneset", ZoneSetData::ToJson(device.data.activeZoneSet));
+        result.SetNamedValue(L"editor-show-spacing", json::value(device.data.showSpacing));
+        result.SetNamedValue(L"editor-spacing", json::value(device.data.spacing));
+        result.SetNamedValue(L"editor-zone-count", json::value(device.data.zoneCount));
+
+        return result;
+    }
+
+    std::optional<DeviceInfoJSON> DeviceInfoJSON::FromJson(const json::JsonObject& device)
+    {
+        try
+        {
+            DeviceInfoJSON result;
+
+            result.deviceId = device.GetNamedString(L"device-id");
+
+            const auto zoneSet = ZoneSetData::FromJson(device.GetNamedObject(L"active-zoneset"));
+            if (zoneSet.has_value())
+            {
+                result.data.activeZoneSet = *zoneSet;
```

Comment:
nit: let's make `zoneSet` mutable and move it instead of copying.

Reply:
Done.